include/envoy/registry/registry.h

```diff
@@ -342,28 +342,28 @@ template <class Base> class FactoryRegistry : public Logger::Loggable<Logger::Id
   static std::unique_ptr<absl::flat_hash_map<std::string, Base*>> buildFactoriesByType() {
     auto mapping = std::make_unique<absl::flat_hash_map<std::string, Base*>>();
 
-    for (const auto& factory : factories()) {
-      if (factory.second == nullptr) {
+    for (const auto& [factory_name, factory] : factories()) {
+      if (factory == nullptr) {
         continue;
       }
 
       // Skip untyped factories.
-      std::string config_type = factory.second->configType();
+      std::string config_type = factory->configType();
       if (config_type.empty()) {
         continue;
       }
 
       // Register config types in the mapping and traverse the deprecated message type chain.
       while (true) {
         auto it = mapping->find(config_type);
-        if (it != mapping->end() && it->second != factory.second) {
+        if (it != mapping->end() && it->second != factory) {
           // Mark double-registered types with a nullptr.
```

Comment:
Do we want to destructure `*it` here?

Reply:
I mean we can assign and give an alias to it->second. But destructing an iterator means that we won't be able to do 
it != end and it->first isn't really used here.

Reply:
also, I believe you can't use structured binding on absl::iterator.

Reply:
You should be able to dereference and destructure as `auto& [first, second] = *it;`. `first` and `second` are just aliases so you can continue checking `it != end`. I agree with you though, it wouldn't really add much here.

Reply:
Oh cool, yeah that makes a lot of sense.